### ISSUE TITLE
Set up finetune script on GCP

### ISSUE THREAD

**Issue by Locke0** [ManifoldRG/software-control#174]
- [x] Dataloading
- [x] Checkpoint saving
- [ ] We plan to run 3 epochs with LoRA, add val set running and logging logic before we start finetuning, and after each epoch
- [ ] See if you can save the train, val loss metrics to w&b (see if the key I shared a while ago works). if too much setup overhead, just save locally in a json or something
- [x] Add time tracking per inference step for finetuning run time and cost estimate
- [ ] Monitor memory (nvitop and nvtop are good)
- [ ] install [tmux](https://tmuxcheatsheet.com/) & run script and nvtop
    - tmux a - reattach
    - ctrl + b + arrow key - swtich window pane
    - ctrl + b, shift + " - create split screen pane
    - ctrl + b, shift + % - same but horizontal?
    - ctrl + b, [ - then you can page up, down
    - ctrl + d, terminate the tmux pane
